document/core/exec/runtime.rst

```diff
@@ -26,8 +30,12 @@ In order to avoid ambiguities, values are therefore represented with an abstract
 It is convenient to reuse the same notation as for the |CONST| :ref:`instructions <syntax-const>` and |REFNULL| producing them.
 
 References other than null are represented with additional :ref:`administrative instructions <syntax-instr-admin>`.
-They either are *function references*, pointing to a specific :ref:`function address <syntax-funcaddr>`,
-or *external references* pointing to an uninterpreted form of :ref:`extern address <syntax-externaddr>` that can be defined by the :ref:`embedder <embedder>` to represent its own objects.
+They either are *scalar references*, containing a 31-bit :ref:`integer <syntax-int>`,
+*structure references*, pointing to a specific :ref:`structure address <syntax-structaddr>`,
+*array references*, pointing to a specific :ref:`array address <syntax-arrayaddr>`,
+*function references*, pointing to a specific :ref:`function address <syntax-funcaddr>`,
+or *host references* pointing to an uninterpreted form of :ref:`host address <syntax-hostaddr>` defined by the :ref:`embedder <embedder>`.
+Any reference can furthermore be wrapped up as an *external reference*.
```

Comment:
Do we need to exclude external references from this last statement or is that unnecessarily pedantic?

Reply:
Reworded.